### PR TITLE
makes plasma stabilizer only care for helmet

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Plasma stabilizer module no longer prevents plasmamen self-combust if you dont have envirosuit/gloves.

image

Also adds a trait for clothes that prevents Plasmamen from igniting, because it was requested to not have snowflake istype checks.

## Why It's Good For The Game

Wearing a single module shouldn't remove an entire feature to playing a certain race, not that the description of it makes any sense. In the past, Plasmamen either needed a full plasma room or hyper nobellium internals. Things that require effort to make and are limited in some way. Wearing a single module on a suit to bypass this entire feature is lame as hell.
Its intended feature was to allow Plasmamen to safely equip MODsuits, which it now does.

## Changelog

:cl:
fix: MODsuit plasma stabilizers no longer protect plasmamen from self-combustion if they aren't wearing an envirosuit or gloves.
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
it's not a fix

**Comment by Likteer** [comment]
Torso? Understandable. Gloves? I like the module allowing you to survive your hand being torn off like old suits.

**Comment by JohnFulpWillard** [comment]
> it's not a fix

It is because it’s making functionally work as intended. Plasma stabilizer is meant to make plasmamen able to safely equip a modsuit by making them safe to remove their helmet, it was just added in lazily.

**Comment by Serxule** [comment]
> > it's not a fix
> 
> It is because it’s making functionally work as intended. Plasma stabilizer is meant to make plasmamen able to safely equip a modsuit by making them safe to remove their helmet, it was just added in lazily.

normal hardsuits did this too, so no.

**Comment by JohnFulpWillard** [comment]
MODsuits are not hardsuits, they replaced the them, that doesn’t mean they are them. Additionally, hardsuits didn’t protect against fire when you didn’t have it on, MODsuits currently do.

**Comment by Colovorat** [comment]
Like the whole thing is just gonna affect 5% of player base who plays plasmeme, and 5% of plasmemes who uses this thing. It's just not worth it. And it actually gives plasmemes access to drip. Yes I added plasmeme hats, but it's just not enough. I would say give it some downside instead of just not working without a suit

**Comment by JohnFulpWillard** [comment]
Giving it other downsides and accepting it as a feature is a balance, not fixing it to work as it’s supposed to. I see this bug exploited constantly and it is very annoying  so i want to fix it. Feel free to balance it yourself in a separate PR.

**Comment by Colovorat** [comment]
The thing you do is balance too. Fikou intended it to be this way, but as it's being "exploited", editing this feature is balance change

**Comment by JohnFulpWillard** [comment]
The intent of the feature is to allow plasmamen to equip MODsuits without burning. That hasn’t been balanced, it’s still doing that. It just removes the exploit that allows plasmamen to remove their need for an envirosuit entirely, not even using it to equip MODsuits.

**Comment by Colovorat** [comment]
Then remove need for gloves at least. Make it require only envirosuit

File: code/modules/mod/modules/modules_general.dm [hunk 1]
```diff
@@ -468,10 +468,10 @@
 ///Plasma Stabilizer - Prevents plasmamen from igniting in the suit
 /obj/item/mod/module/plasma_stabilizer
 	name = "MOD plasma stabilizer module"
-	desc = "This system essentially forms an atmosphere of its' own inside the suit, \
-		safely ejecting oxygen from the inside and allowing the wearer, a plasmaman, \
-		to have their internal plasma circulate around them somewhat like a sauna. \
-		This prevents them from self-igniting, and leads to greater comfort overall. \
+	desc = "This system essentially forms an atmosphere of its own, within the suit, \
+		efficiently and quickly preventing oxygen from causing the user's head to burst. \
```
Comment (ATH1909) [inline code/modules/mod/modules/modules_general.dm:472]:
```suggestion
		efficiently and quickly preventing oxygen from causing the user's head to burst into flame. \
```

File: code/modules/mod/modules/modules_general.dm [hunk 2]
```diff
@@ -468,10 +468,10 @@
...
+		This allows plasmamen to safely remove their helmet, allowing for easier \
+		equipping of any MODsuit related equipment, or otherwise. \
```
Comment (ATH1909) [inline code/modules/mod/modules/modules_general.dm:474]:
```suggestion
		equipping of any MODsuit-related equipment, or otherwise. \
```

**Comment by Serxule** [comment]
> MODsuits are not hardsuits, they replaced them, that doesn’t mean they are them. Additionally, hardsuits didn’t protect against fire when you didn’t have it on, MODsuits currently do. This is ignoring the fact they do have the functionality you're talking about , when the MODsuit is ON.

maybe you should clarify what you're talking about then, because the way i read this pr is "plasmaman will burn in modsuits if naked"

**Comment by Fikou** [comment]
should allow no gloves probably

**Comment by MrMelbert** [comment]
I'm interested in this pr, but I the plasmaman self ignition code is so ugly I'm not sure if it should be further complicated in its current state

**Comment by JohnFulpWillard** [comment]
> I'm interested in this pr, but I think the plasmaman self ignition code is so ugly, that I'm not sure if it should be further complicated in its current state.
> 
> ~I wish that they checked for a clothing trait or something instead of hard checking istypes of plasmamen clothes. That'd make it a bit easier to flip on or off plasmaman protection for modsuits or w/e~

This isnt conplicating it further than adding trait checks, I just made them do several checks instead of one large text

**Pull request merged**